core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -279,8 +287,13 @@ private[spark] object ReliableCheckpointRDD extends Logging {
 
     // Register an on-task-completion callback to close the input stream.
     context.addTaskCompletionListener(context => deserializeStream.close())
-
-    deserializeStream.asIterator.asInstanceOf[Iterator[T]]
+    Utils.tryWithSafeFinally {
```

Comment:
I don't think you can close it here, right? you're returning an iterator on the stream